src/socket.c

```diff
@@ -2274,6 +2274,40 @@ int _UDP6_listen (Socket *socket, const void *host, WORD port)
 }
 #endif  /* USE_IPV6 */
 
+/*
+ * Check progress on non-blocking SOCK_STREAM connection.
+ * Only called on 2nd (3rd etc) time a non-blocking connect() is called.
+ * Also checked from select_s().
+ */
+int _sock_chk_connect (Socket *socket)
+{
+  if (socket->so_error != EALREADY)
+    return (socket->so_error);
+
+  if (socket->so_state & SS_ISCONNECTED)
+    return (0);
+
+  if (socket->so_state & SS_CONN_REFUSED)
+    return (ECONNREFUSED);
+
+  if (chk_timeout(socket->nb_timer))
```

Comment:
For some reason this timer expires immediately.

Reply:
Oh I see.  `socket->delay` is set to 0 in a non-blocking socket.  So `socket->nbtimer` should be set from the default `sock_delay`.